src/samples/Shared/AspNetExtensions.cs

```diff
@@ -64,12 +65,12 @@ public static class AspNetExtensions
     /// `AzureBotServiceTokenHandling` defaults to true and should always be true until Azure Bot Service sends Entra ID token.
     /// `AllowedCallers` is optional and defaults to "*".  Otherwise, a list of AppId's the Agent will accept requests from.
     /// </remarks>
-    public static void AddAgentAspNetAuthentication(this IServiceCollection services, IConfiguration configuration, string tokenValidationSectionName = "TokenValidation", ILogger logger = null)
+    public static void AddAgentAspNetAuthentication(this IServiceCollection services, IConfiguration configuration, string tokenValidationSectionName = "TokenValidation", ILogger logger = null!)
```

Comment:
Where you getting possible null warnings in this function?   curious why you added the !

Reply:
The project I added, uses the default settings for `nullable`, and it produced several warnings.

As this file is meant to be shared to other projects, and customers will copy/paste, I think it should be free of warnings with the VS defaults.